City-of-Helsinki/drupal-helfi-rekry
# UHF-8706

# [UHF-8706](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8706)

- Send indexing request to google when a job listing is published.
- A temporary redirect is created for the job listing before sending the request
- Sending deindexing request to google when a job listing is unpublished
- The temporary redirect is removed after the request.

## What was done
- Google api integration using google's integration library
- Read the documentation

## How to install
* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-8706`
  * `make fresh`
* Run `make drush-cr`
* You must set the api key. It can just be for example empty json object as string `'{}'`
  * local.settings.php: `$config['helfi_google_api.settings']['indexing_api_key'] = 'key-goes-here'`
  * Even with correct api-key, local environment will get an error from API
  * The api key is set to production environment key vault.


## How to test
You might want to use xdebug:
- Set breakpoint in GoogleApi.php
  - Lines 103 and 138, the line where the request is executed.

You can try either by drush-command OR by running drush cron after changing the scheduled date from one of the job listings.

When sending an indexing request:
- See the url which is about to be sent to google api
  - it should be the url to the job listing + `-{timestamp}`
  - If you execute the query, you should see an api error saying that cannot resolve the owner of the url.
  - If you go to edit the job listing in admin-ui, you should see that a redirect has been created for the job listing

When sending an deindexing request
- Follow the indexing-instructions (run drush command or fiddle with the scheduler etc.)
- See that the url matches the redirect
- See that the redirect is deleted after the operation is done.


A command to check url indexing status:
You can try sending a status request by running drush-command. One real url has been indexed and deindexed using the api. you can check the status by running the following drush command:
`drush helfi:google-entity-index-status https://hel.fi/fi/avoimet-tyopaikat/avoimet-tyopaikat/sotepe-02-353-24`

## Continuous documentation




[UHF-8706]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8706?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
